packages/analytics/amplify_analytics_pinpoint_dart/lib/src/analytics_plugin_impl.dart

```diff
@@ -138,15 +146,33 @@ class AmplifyAnalyticsPinpointDart extends AnalyticsPluginInterface {
     );
 
     // Retrieve Unique ID
-    final savedFixedEndpointId =
-        await _endpointInfoStore.read(key: endpointIdStorageKey);
-    final fixedEndpointId = savedFixedEndpointId ?? const Uuid().v1();
-    if (savedFixedEndpointId == null) {
+    final endpointInformationVersion =
+        await _endpointInfoStore.read(key: _endpointInformationVersionKey);
+    String? retrievedEndpointId;
```

Comment:
I think you can likely make this final an non-nullable. See other comments below.